sphinx/ext/graphviz.py

```diff
@@ -286,24 +286,26 @@ def render_dot_latex(self, node, code, options, prefix='graphviz'):
         raise nodes.SkipNode
 
     is_inline = self.is_inline(node)
-    if is_inline:
-        para_separator = ''
-    else:
-        para_separator = '\n'
 
-    if fname is not None:
-        post = None  # type: unicode
-        if not is_inline and 'align' in node:
+    if not is_inline:
+        pre = ''
+        post = ''
+        if 'align' in node:
             if node['align'] == 'left':
-                self.body.append('{')
-                post = '\\hspace*{\\fill}}'
+                pre = '{'
+                post = r'\hspace*{\fill}}'
             elif node['align'] == 'right':
-                self.body.append('{\\hspace*{\\fill}')
+                pre = r'{\hspace*{\fill}'
                 post = '}'
-        self.body.append('%s\\includegraphics{%s}%s' %
-                         (para_separator, fname, para_separator))
-        if post:
-            self.body.append(post)
+            elif node['align'] == 'center':
+                pre = r'{\hfill'
+                post = r'\hspace*{\fill}}'
+        self.body.append('\n%s' % pre)
+
+    self.body.append(r'\includegraphics{%s}' % fname)
+
+    if not is_inline:
+        self.body.append('%s\n' % post)
```

Comment:
This looks perfectly legit, but I think it may be simplified like this:

```
    if not is_inline:
        pre = ''
        post = ''
        if 'align' in node:
            if node['align'] == 'left':
                pass
            elif node['align'] == 'right':
                pre = r'\hfill'
            elif node['align'] == 'center':
                pre = r'\hfill'
                post = r'\hfill{}'
        self.body.append('\n%s' % pre)
```

Indeed left alignment is default (I don't think Sphinx has mark-up for flushright environment) hence the ``pass``  for left alignment.

And I removed one pair of braces in all cases.

However I am not familiar with graphviz, so I may be missing something. Does the above work well with your documents? It is important that a blank line above and another one below separates this from other typeset contents.

Reply:
It might comes from `visit_image()` of LaTeXTranslator.
https://github.com/sphinx-doc/sphinx/blob/1.6.3/sphinx/writers/latex.py#L1738

Reply:
indeed @tk0miya. The code is ok, braces `{` and `}` seem superfluous. Perhaps it is better to use indeed `\hspace` which is LaTeX command rather than `\hfill` which is the underlying TeX primitive which ultimately gets "exectued". Anyway, a little refactoring of `visit_image()` is needed due to new issue #4085 which I assigned to myself and discovered now from looking at its code.

Reply:
I thought I had understood a reason for extra braces in `visit_image()` code https://github.com/sphinx-doc/sphinx/blob/1.6.3/sphinx/writers/latex.py#L1738: they would be needed indeed to hide the square brackets `[ ...]` of `\includegraphics` options in case of an image used as a list heading, say from this input

```rst
|picture|
  description of item
```

But trying to use ``:align: center`` option gives anyhow ``WARNING: Error in "image" directive: "center" is not a valid value for the "align" option within a substitution definition.  Valid values for "align" are: "top", "middle", "bottom".``.

Only non-inline images will use the code path for ``:align:`` center, left or right. Thus I am still looking for reason of extra braces. (This is not much important)!

I don't think one needs `\hspace*{\fill}` but `\hspace{\fill}` works in my brief testing, however I may have missed something ...